### PR TITLE
[MBL-17983][All] New Quiz - Dedicated WebView

### PR DESCRIPTION
**Pull request by tamaskozmer**
Some notes:
- There are lot of changes in this PR, I refactored the LTI launching logic, moved all the code into the common package and now all the 3 apps use the same code for LTI launching.
- This PR fixed an other issue as well which I discovered in the process. When you launch an LTI submission from the Student submission details screen it was only possible to launch it once. If the users closed the LTI tool and tried to launch it again it failed. This was due to passing an already authenticated url to the submission details screen. I had to refactor this as well.
- Currently only the new quizzes open internally but other LTI tools can be easily integrated.

Test plan:
- Verify that the new quizzes LTI opens internally and the Return button works as intended.
- Smoke test all the LTI related features (opening from nav bar, opening LTI tabs, opening and submitting LTI assignments, opening LTIs from HTML content) The new Parent app only has the first one.
- Check that the Teacher app embeds the LTI submissions.

refs: MBL-17983
affects: Student, Teacher
release note: Better new quizzes integration

## Checklist

- [ ] Run E2E test suite
- [x] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/SpeedGraderLtiSubmissionFragment.kt [hunk 1]
```diff
@@ -53,19 +53,30 @@ class SpeedGraderLtiSubmissionFragment : Fragment() {
...
+
+            override fun onPageStartedCallback(webView: WebView, url: String) {
+                if (isAdded) binding.webViewProgress.setVisible()
+            }
+
+            override fun onPageFinishedCallback(webView: WebView, url: String) {
+                if (isAdded) binding.webViewProgress.setGone()
+            }
+
+            override fun canRouteInternallyDelegate(url: String): Boolean = false
```
Comment (hermannakos) [inline apps/teacher/src/main/java/com/instructure/teacher/fragments/SpeedGraderLtiSubmissionFragment.kt:69]: We sure we don't want to route anywhere from here? Edit: see other comment

**Review by hermannakos — CHANGES_REQUESTED** [review]
- [x] In app navigation is not enabled from the new screen. It will navigate inside the webview. 

- [x] When navigating inside the webview pressing back pops the fragment and not the webview backstack.

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/SpeedGraderLtiSubmissionFragment.kt (hunk 1 above)
Comment (tamaskozmer) [reply apps/teacher/src/main/java/com/instructure/teacher/fragments/SpeedGraderLtiSubmissionFragment.kt:69]: I don't think we need to route from LTI tools. These are completely separate tools, outside of Canvas, it was previously opened in Custom Tabs, so we didn't have any in app routing, and as far as I know this wasn't a requirement.

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by hermannakos — APPROVED** [review]
QA + 1

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Review by kristofnemere — APPROVED** [review]

**Pull request merged**